### PR TITLE
check for Transformers creating empty derivatives, and throw an error…

### PR DESCRIPTION
**Pull request by bcail**
… if it happens

### CODE REVIEW

File: tests/webapp_t.py [hunk 1]
```diff
@@ -618,6 +619,13 @@ def _assert_tmp_has_no_files(self):
         any_files = any([path.isfile(path.join(tmp, n)) for n in listdir(tmp)])
         self.assertTrue(not any_files, "There are too many files in %s: %s" % (tmp, any_files))
 
+    @patch('loris.transforms.KakaduJP2Transformer.transform')
+    def test_empty_image_file_error(self, mock_transform):
+        to_get = '/%s/full/300,/0/default.jpg' % (self.test_jp2_color_id,)
+        resp = self.client.get(to_get)
+        self.assertEqual(resp.status_code, 500)
+        self.assertTrue('empty derivative file' in resp.data.decode('utf8'))
```
Comment (alexwlchan) [inline tests/webapp_t.py]: I'm not sure we should be exposing this error externally; just logging it somewhere.

**Review by alexwlchan — COMMENTED** [review]
I'm curious – what are the circumstances when this could happen? Is it something you've seen in the wild?

File: tests/webapp_t.py (hunk 1 above)
Comment (bcail) [reply tests/webapp_t.py]: @alexwlchan should that works for me. I could make just a general server side error.

**Comment by bcail** [comment]
@alexwlchan yes, I saw it in our Loris this week. It can happen if there's a problem with the transform that doesn't raise an exception (note: we might need to improve error checking in transform()). Our issue might have been caused by a missing src image file, or something different related to cache moving/pruning.

File: tests/webapp_t.py (hunk 1 above)
Comment (bcail) [reply tests/webapp_t.py]: done - pushed a new commit.

**Comment by bcail** [comment]
@alexwlchan ping. Are you good with this now, or do you want to discuss more?

**Comment by alexwlchan** [comment]
Would be good to improve error checking in the transform, but happy with this as-is.

**Pull request merged**